discovery-provider/src/queries/query_helpers.py

```diff
@@ -102,132 +103,44 @@ def populate_user_metadata(session, user_ids, users, current_user_id, with_track
-            Follow.is_current == True,
-            Follow.is_delete == False,
-            Follow.followee_user_id.in_(user_ids)
-        )
-        .group_by(Follow.followee_user_id)
-        .all()
-    )
-    follower_count_dict = {user_id: follower_count for (
-        user_id, follower_count) in follower_counts}
-
-    # build dict of user id --> followee count
-    followee_counts = (
-        session.query(
-            Follow.follower_user_id,
-            func.count(Follow.follower_user_id)
-        )
-        .filter(
-            Follow.is_current == True,
-            Follow.is_delete == False,
-            Follow.follower_user_id.in_(user_ids)
-        )
-        .group_by(Follow.follower_user_id)
-        .all()
-    )
-    followee_count_dict = {user_id: followee_count for (
-        user_id, followee_count) in followee_counts}
-
-    # build dict of user id --> repost count
-    repost_counts = (
-        session.query(
-            Repost.user_id,
-            func.count(Repost.user_id)
-        )
-        .filter(
-            Repost.is_current == True,
-            Repost.is_delete == False,
-            Repost.user_id.in_(user_ids)
-        )
-        .group_by(Repost.user_id)
-        .all()
-    )
-    repost_count_dict = {user_id: repost_count for (
-        user_id, repost_count) in repost_counts}
-    track_save_count_dict = {}
-    if with_track_save_count:
-        # build dict of user id --> track save count
-        track_save_counts = (
-            session.query(
-                Save.user_id,
-                func.count(Save.user_id)
-            )
-            .filter(
-                Save.is_current == True,
-                Save.is_delete == False,
-                Save.save_type == SaveType.track,
-                Save.user_id.in_(user_ids)
-            )
-            .group_by(Save.user_id)
-            .all()
-        )
-        track_save_count_dict = {user_id: user_track_save_count for (
-            user_id, user_track_save_count) in track_save_counts}
+    # build dict of user id --> track/playlist/album/follower/followee/repost/track save counts
+    count_dict = {
+        user_id: {
+            response_name_constants.track_count: track_count,
+            response_name_constants.playlist_count: playlist_count,
+            response_name_constants.album_count: album_count,
+            response_name_constants.follower_count: follower_count,
+            response_name_constants.followee_count: following_count,
+            response_name_constants.repost_count: repost_count,
+            response_name_constants.track_save_count: track_save_count
+        } for (
+            user_id,
+            track_count,
+            playlist_count,
+            album_count,
+            follower_count,
+            following_count,
+            repost_count,
+            track_save_count
+        ) in aggregate_user
+    }
 
     # build dict of user id --> track blocknumber
     track_blocknumbers = (
```

Comment:
Wonder if it would be useful to include blocknumber in the matview? WDYT

Reply:
i think this is a little dangerous because the block number here might drift from the block number on the root table itself and we do want this block number to be as fresh as possible for syncs. the stats on the matview might not be as fresh as we want. i think it's worth adding to the matview, but maybe as a follow-on to this PR

Reply:
Just seems unfortunate to have to do an extra roundtrip for this here. Do we need <5s freshness for syncs?

Reply:
https://github.com/AudiusProject/audius-protocol/blob/master/creator-node/src/middlewares.js#L266-L271

all this code is actually going away in a few weeks once replica sets are backfilled and available on UserReplicaSetManager. basically we no longer need to use discovery from cnodes to verify primary. that can happen directly from chain